test/test-jsdomparser.js

```diff
@@ -190,6 +195,19 @@ describe("Test JSDOM functionality", function() {
       }
     }
   });
+
+  it("should handle encoding HTML correctly", function() {
+    var baseStr = "<p>Hello, everyone &amp; all their friends, &lt;this&gt; is a &quot; test.</p>";
+    var doc = new JSDOMParser().parse(baseStr);
+    var p = doc.getElementsByTagName("p")[0];
+    expect("<p>" + p.innerHTML + "</p>").eql(baseStr);
+
+    p.textContent = p.textContent + " ";
+    expect("<p>" + p.innerHTML + "</p>").eql(baseStr.replace("</p>", " </p>"));
```

Comment:
So JSDOMParser actually stores `innerHTML` straight from the parser and then regurgitates it for the first assertion.

Assigning to `textContent` will break that cache and force it to re-encode from the escaped version.